tictactoe/lib/tictactoe/game.ex

```diff
@@ -93,19 +116,34 @@ defmodule Tictactoe.Game do
   defp has_win([mark, mark, mark]), do: true
   defp has_win(_),                  do: false
 
-  defp switch_players(game = %Tictactoe.Game{ game_state: "playing" }) do
-    Map.put(game, :current_player, next_player(game))
+  defp switch_players(game = %Game{ game_state: "playing" }) do
+    %{ game | current_player: next_player(game) }
   end
 
   defp switch_players(game) do
     game
   end
 
-  defp next_player(%Tictactoe.Game{ current_player: "X" }) do
-    "O"
+  defp next_player(game) do
+    case game.current_player do
+      %{"mark" => "X"} -> game.player_o
```

Comment:
Could it be something like that `game.player_x => game.player_o`? It feels like it's an inappropriate Intimacy - does the game/next_player function need to know about the data of a player?

Reply:
I did try that, but Elixir doesn't allow me to call `game.player_x` inside the condition. :(